src/dymaptic.GeoBlazor.Core/Components/Layers/WCSLayer.cs

```diff
@@ -0,0 +1,147 @@
+﻿using dymaptic.GeoBlazor.Core.Components.Renderers;
+using dymaptic.GeoBlazor.Core.Exceptions;
+using Microsoft.AspNetCore.Components;
+using System.Security.Cryptography.X509Certificates;
+using System.Text.Json.Serialization;
+
+
+namespace dymaptic.GeoBlazor.Core.Components.Layers;
+
+/// <summary>
+///     WCS presents raster data from a OGC Web Coverage Service. Raster data are projected and rendered on the client-side.
+///     It supports versions 1.0.0, 1.1.0, 1.1.1, 1.1.2 and 2.0.1. For version 2.0.1, it supports servers that support 
+///     GEOTIFF coverage and implements the following extensions: Scaling, Interpolation, Range Subsetting, CRS, and KVP/Get.
+///     <a target="_blank" href="https://developers.arcgis.com/javascript/latest/api-reference/esri-layers-WCSLayer.html">
+///         ArcGIS
+///         JS API
+///     </a>
+/// </summary>
+public class WCSLayer : Layer
+{
+    /// <inheritdoc />
+    [JsonPropertyName("type")]
+    public override string LayerType => "wcs";
+
+    /// <summary>
+    ///     Constructor for use as a razor component
+    /// </summary>
+    public WCSLayer()
+    {
+    }
+    /// <summary>
+    ///     Constructor for use in code
+    /// </summary>
+    /// <param name="url">
+    ///     The url for the WCS Layer source data.
+    /// </param>
+
+    public WCSLayer(string? url = null, PortalItem? portalItem = null)
```

Comment:
Add `List<DimensionalDefinition> multidimensionalDefinition, RasterStretchRenderer renderer, double opacity, string title` to this constructor. Use the constructor in your `WCSLayers.razor` samples page instead of setting these properties directly. In fact, I think we should make these `private set;` properties (only for the ones that are not also `[Parameter]`), because if you set them _after_ the layer is rendered, it will break.